### ISSUE TITLE
kubectl -s and --server behave differently

### ISSUE THREAD

**Issue by jasonkuhrt** [GoogleCloudPlatform/kubernetes#2862]
```sh
$ kubectl --insecure-skip-tls-verify=true -s=https://xxx.xxx.xx.xx version                                                                                            

F1211 10:42:31.611421    9965 cmd.go:215] Error creating kubernetes client: host must be a URL or a host:port pair: =https://xxx.xxx.xx.xx
```

```sh
$ kubectl --insecure-skip-tls-verify=true --server=https://xxx.xxx.xx.xx version                                                                                        

Client Version: version.Info{Major:"0", Minor:"5", GitVersion:"v0.5", GitCommit:"e2e1c87ae06b303af92fdb5f44af60643d80469e", GitTreeState:"clean"}
Server Version: &version.Info{Major:"0", Minor:"4+", GitVersion:"v0.4.2", GitCommit:"2902951c445da6e6ead9e406b3c8e19bc28c2b4a", GitTreeState:"clean"}
```

-- edit

To be clear, the commands should output the same thing but actually the `-s` version returns a certain formatting error.

**Comment by wojtek-t** [comment]
It seems that it's not an issue with kubectl itself, right?

**Comment by bgrant0607** [comment]
Correct.

#4014 is the same issue.

**Comment by satnam6502** [comment]
Is this not because we have switched to a different flag library?

**Comment by brendandburns** [comment]
Jbeda has committer rights on pflag, so we should just fix these things...

Brendan

**Comment by filbranden** [comment]
I can look into it. In my experience I saw `-t <argument>` work in another context so I'm not sure why it's failing here... I also have commit rights to `spf13/pflag` so if it's a bug there I can report it and try to fix it upstream.

**Comment by filbranden** [comment]
This is awful...

`kubectl -s https://w.x.y.z version` doesn't work but `kubectl version -s https://w.x.y.z` does.

The problem seems to be that spf13/cobra is trying to find the command (`version` in this case) before parsing flags, so in effect it has code to "guess" what the flags are (anything starting with a `-`) and "skip" them to find the command. In this case, it's not skipping the `https://w.x.y.z` and it thinks that's actually the command.

Not sure if it's an easy fix... But I'll keep looking at it.

Overall, I'm very appalled at flag handling in Golang, both built-in as the third-party ones... Why did they have to reinvent the wheel??? And do a terrible job at it?

**Comment by brendandburns** [comment]
See https://github.com/spf13/cobra/pull/66 for a fix to upstream.

**Comment by brendandburns** [comment]
Closed by #5464

**Comment by pmorie** [comment]
@brendandburns I was looking at P1 issues today and it looks like you meant to close this one out.
